x-pack/plugins/security/server/deprecations/privilege_deprecations.test.ts

```diff
@@ -0,0 +1,156 @@
+    const authz = authorizationService.setup({
+      http: mockCoreSetup.http,
+      capabilities: mockCoreSetup.capabilities,
+      getClusterClient,
+      license: mockLicense,
+      loggers: loggingSystemMock.create(),
+      kibanaIndexName,
+      packageVersion: 'some-version',
+      buildNumber: 42,
+      features: mockFeaturesSetup,
+      getSpacesService: mockGetSpacesService,
+      getCurrentUser: jest.fn(),
+    });
+
+    const { getKibanaRolesByFeatureId } = getPrivilegeDeprecationsServices(authz, mockLicense);
+
+    it('happy path, we should have the attribute roles', async () => {
+      const apiResponse = async () => ({
+        first_role: {
+          cluster: [],
+          indices: [],
+          applications: [
+            {
+              application,
+              privileges: ['feature_siem.all', 'feature_siem.cases_read'],
+              resources: ['space:securitySolutions'],
+            },
+          ],
+          run_as: [],
+          metadata: {
+            _reserved: true,
+          },
+          transient_metadata: {
+            enabled: true,
+          },
+        },
+      });
+
+      mockAsCurrentUser.asCurrentUser.security.getRole.mockImplementation((async () => ({
+        body: await apiResponse(),
+      })) as any);
+
+      const mockContext = {
+        esClient: mockAsCurrentUser,
+        savedObjectsClient: jest.fn(),
+      } as unknown as GetDeprecationsContext;
+
+      const resp = await getKibanaRolesByFeatureId({ context: mockContext, featureId: 'siem' });
+      expect(resp).toMatchInlineSnapshot(`
+        Object {
+          "roles": Array [
+            Object {
+              "_transform_error": Array [],
+              "_unrecognized_applications": Array [],
+              "elasticsearch": Object {
+                "cluster": Array [],
+                "indices": Array [],
+                "run_as": Array [],
+              },
+              "kibana": Array [
+                Object {
+                  "base": Array [],
+                  "feature": Object {
+                    "siem": Array [
+                      "all",
+                      "cases_read",
+                    ],
+                  },
+                  "spaces": Array [
+                    "securitySolutions",
+                  ],
+                },
+              ],
+              "metadata": Object {
+                "_reserved": true,
+              },
+              "name": "first_role",
+              "transient_metadata": Object {
+                "enabled": true,
+              },
+            },
+          ],
+        }
+      `);
+    });
+
+    it('unhappy path, we should have the attribute errros', async () => {
+      const apiResponse = async () => {
+        throw new Error('It happens');
+      };
+
+      mockAsCurrentUser.asCurrentUser.security.getRole.mockImplementation((async () => ({
+        body: await apiResponse(),
+      })) as any);
```

Comment:
Nit: could be simplified
```suggestion
      mockAsCurrentUser.asCurrentUser.security.getRole.mockRejectedValue(new Error('It happens'));
```